robot/robot.py

```diff
@@ -37,22 +38,33 @@ def __init__(self, robotd_path=ROBOTD_ADDRESS):
             self.known_gamestates,
         ]
 
+        self._display_boards()
         self._wait_for_power_board()
-
-    def _wait_for_power_board(self):
-        power_boards = self.power_boards
-        if not power_boards:
-            raise RuntimeError('Cannot find Power Board!')
-
-        print('Turning on power board.')
+        self.wait_start()
         self.power_board.power_on()
+        print("Starting user code.")
+
+    def _display_boards(self):
+        print("Found the following hardware devices:")
+        for board_list in self.all_known_boards:
+            for board in board_list:
+                print(str(board))  # Force string representation
 
+    def wait_start(self):
```

Comment:
I wonder if we should push this down into the `PowerBoard` class?

Reply:
Hmm, interesting idea, i'll see how easy it is to migrate the code